test/integration/gardenlet/managedseed/managedseed_test.go

```diff
@@ -0,0 +1,357 @@
+				g.Expect(testClient.Get(ctx, client.ObjectKey{Name: "gardenlet", Namespace: gardenNamespace.Name}, &corev1.Service{})).To(Succeed())
+				g.Expect(testClient.Get(ctx, client.ObjectKey{Name: "gardenlet", Namespace: gardenNamespace.Name}, &corev1.ServiceAccount{})).To(Succeed())
+				g.Expect(testClient.Get(ctx, client.ObjectKey{Name: "gardenlet", Namespace: gardenNamespace.Name}, &appsv1.Deployment{})).To(Succeed())
+			}).Should(Succeed())
+		}
+	)
+
+	BeforeEach(func() {
+		Eventually(func(g Gomega) {
+			g.Expect(mgrClient.Get(ctx, client.ObjectKeyFromObject(gardenNamespace), &corev1.Namespace{})).To(Succeed())
+		}).Should(Succeed())
+
+		gardenletConfig, err := encoding.EncodeGardenletConfiguration(&gardenletconfigv1alpha1.GardenletConfiguration{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: gardenletconfigv1alpha1.SchemeGroupVersion.String(),
+				Kind:       "GardenletConfiguration",
+			},
+			GardenClientConnection: &gardenletconfigv1alpha1.GardenClientConnection{
+				KubeconfigSecret: &corev1.SecretReference{
+					Name:      "gardenlet-kubeconfig",
+					Namespace: gardenNamespace.Name,
+				},
+			},
+			SeedConfig: &gardenletconfigv1alpha1.SeedConfig{
+				SeedTemplate: gardencorev1beta1.SeedTemplate{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{testID: testRunID},
+						Annotations: map[string]string{
+							"bar": "baz",
+						},
+					},
+					Spec: gardencorev1beta1.SeedSpec{
+						Backup: &gardencorev1beta1.SeedBackup{
+							Provider: "test",
+							Region:   pointer.String("bar"),
+							SecretRef: corev1.SecretReference{
+								Name:      "test-backup-secret",
+								Namespace: gardenNamespace.Name,
+							},
+						},
+						SecretRef: &corev1.SecretReference{
+							Name:      "test-seed-secret",
+							Namespace: gardenNamespace.Name,
+						},
+						DNS: gardencorev1beta1.SeedDNS{
+							IngressDomain: pointer.String("someingress.example.com"),
+						},
+					},
+				},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		managedSeed = &seedmanagementv1alpha1.ManagedSeed{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "managedseed-",
+				Namespace:    gardenNamespace.Name,
+				Labels:       map[string]string{testID: testRunID},
+			},
+			Spec: seedmanagementv1alpha1.ManagedSeedSpec{
+				Gardenlet: &seedmanagementv1alpha1.Gardenlet{
+					Deployment: &seedmanagementv1alpha1.GardenletDeployment{
+						ReplicaCount:         pointer.Int32(1),
+						RevisionHistoryLimit: pointer.Int32(1),
+						Image: &seedmanagementv1alpha1.Image{
+							PullPolicy: pullPolicyPtr(corev1.PullIfNotPresent),
+						},
+						VPA: pointer.Bool(false),
+					},
+					Config:    *gardenletConfig,
+					Bootstrap: bootstrapPtr(seedmanagementv1alpha1.BootstrapToken),
+				},
+			},
+		}
+
+		shoot = &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "shoot" + testRunID,
```

Comment:
Since you use this name also in `suite_test.go`, probably export it in a variable there and reuse it here.